spark-bigquery-dsv2/spark-3.4-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark34IndirectWriteIntegrationTest.java

```diff
@@ -30,6 +48,42 @@ public void setParquetLoadBehaviour() {
     spark.conf().set("enableListInference", "true");
   }
 
+  private TableResult insertAndGetTimestampNTZToBigQuery(LocalDateTime time, String format)
+      throws InterruptedException {
+    List<Row> rows = Arrays.asList(RowFactory.create(time));
+    Dataset<Row> df =
+        spark.createDataFrame(
+            rows,
+            new StructType(
+                new StructField[] {
+                  StructField.apply("foo", DataTypes.TimestampNTZType, true, Metadata.empty())
+                }));
+    writeToBigQuery(df, SaveMode.Overwrite, format);
+    BigQuery bigQuery = IntegrationTestUtils.getBigquery();
+    TableResult result =
+        bigQuery.query(
+            QueryJobConfiguration.of(String.format("Select foo from %s", fullTableName())));
+    return result;
+  }
+
   // tests from superclass
+  @Test
+  public void testTimestampNTZWriteToBigQueryAvroFormat() throws InterruptedException {
+    LocalDateTime time = LocalDateTime.of(2023, 9, 1, 12, 23, 34, 268543 * 1000);
+    TableResult result = insertAndGetTimestampNTZToBigQuery(time, "avro");
+    assertThat(result.getSchema().getFields().get(0).getType())
+        .isEqualTo(LegacySQLTypeName.DATETIME);
+    assertThat(result.streamValues().findFirst().get().get(0).getValue().toString())
+        .isEqualTo(time.toString());
+  }
 
+  @Test
+  public void testTimestampNTZWriteToBigQueryParquetFormat() throws InterruptedException {
+    LocalDateTime time = LocalDateTime.of(2023, 9, 15, 12, 36, 34, 268543 * 1000);
+    TableResult result = insertAndGetTimestampNTZToBigQuery(time, "parquet");
+    assertThat(result.getSchema().getFields().get(0).getType())
+        .isEqualTo(LegacySQLTypeName.DATETIME);
+    assertThat(result.streamValues().findFirst().get().get(0).getValue().toString())
+        .isEqualTo(time.toString());
```

Comment:
Please replace time.toString() with the raw string that is expected.

Reply:
The string expected here is the time in string format. What's the benefit of using a string literal hardcoded with the same value as time.toString() here?